jadx-gui/src/main/java/jadx/gui/ui/hexviewer/HexPreviewPanel.java

```diff
@@ -0,0 +1,329 @@
+			BinaryData contentData = hexCodeArea.getContentData();
+			contentData.copyToArray(dataPosition, valuesCache, 0, availableData);
+			if (availableData < CACHE_SIZE) {
+				Arrays.fill(valuesCache, availableData, CACHE_SIZE, (byte) 0);
+			}
+		}
+
+		inspector.setOffset((int) dataPosition);
+	}
+
+	private void createActions() {
+		cutAction = new JMenuItem(NLS.str("popup.cut"));
+		cutAction.addActionListener(e -> performCut());
+
+		copyAction = new JMenuItem(NLS.str("popup.copy"));
+		copyAction.addActionListener(e -> performCopy());
+
+		copyHexAction = new JMenuItem(NLS.str("popup.copy_as_hex"));
+		copyHexAction.addActionListener(e -> performCopyAsCode());
+
+		copyStringAction = new JMenuItem(NLS.str("popup.copy_as_string"));
+		copyStringAction.addActionListener(e -> performCopy());
+
+		pasteAction = new JMenuItem(NLS.str("popup.paste"));
+		pasteAction.addActionListener(e -> performPaste());
+
+		deleteAction = new JMenuItem(NLS.str("popup.delete"));
+		deleteAction.addActionListener(e -> {
+			if (!isEditable()) {
+				performDelete();
+			}
+		});
+
+		selectAllAction = new JMenuItem(NLS.str("popup.select_all"));
+		selectAllAction.addActionListener(e -> performSelectAll());
+
+		copyOffsetItem = new JMenuItem(NLS.str("popup.copy_offset"));
+		copyOffsetItem.addActionListener(e -> copyOffset());
+	}
+
+	private void createPopupMenu() {
+		boolean isEditable = isEditable();
+		popupMenu = new JPopupMenu();
+		popupMenu.add(copyAction);
+
+		if (isEditable) {
+			popupMenu.add(cutAction);
+			popupMenu.add(pasteAction);
+			popupMenu.add(deleteAction);
+			popupMenu.addSeparator();
+		}
+
+		JMenu copyMenu = new JMenu(NLS.str("popup.copy_as"));
+		copyMenu.add(copyHexAction);
+		copyMenu.add(copyStringAction);
+		popupMenu.add(copyMenu);
+		popupMenu.add(copyOffsetItem);
+		popupMenu.add(selectAllAction);
+	}
+
+	private void updatePopupActionStates() {
+
+		boolean selectionExists = isSelection();
+		boolean isEditable = !isEditable();
+
+		cutAction.setEnabled(isEditable && selectionExists);
+		copyAction.setEnabled(selectionExists);
+		copyHexAction.setEnabled(selectionExists);
+		copyStringAction.setEnabled(selectionExists);
+		deleteAction.setEnabled(isEditable && selectionExists);
+
+		selectAllAction.setEnabled(hexCodeArea.getDataSize() > 0);
+	}
+
+	public SectCodeArea getEditor() {
+		return this.hexCodeArea;
+	}
+
+	public HexEditorHeader getHeader() {
+		return this.header;
+	}
+
+	public HexInspectorPanel getInspector() {
+		return this.inspector;
+	}
+
+	public HexSearchBar getSearchBar() {
+		return this.searchBar;
+	}
+
+	public void showSearchBar() {
+		searchBar.showAndFocus();
+	}
+
+	public void performCut() {
+		hexCodeArea.cut();
+	}
+
+	public void performCopy() {
+		hexCodeArea.copy();
+	}
+
+	public void performCopyAsCode() {
+		((CodeAreaOperationCommandHandler) hexCodeArea.getCommandHandler()).copyAsCode();
```

Comment:
This cast here should be to "DefaultCodeAreaCommandHandler" instead.
This will also allow to remove "org.exbin.bined:bined-operation-swing" library dependency. (Operation library adds undo-redo support, which I assume is not needed for JADX)

Also if you would replace all uses of "section.SectCodeArea" class with "basic.CodeArea", you could drop dependency on "org.exbin.bined:bined-swing-section" as you don't use theme/layout capabilities of this extension.

Reply:
> you don't use theme/layout capabilities

```
public SectionCodeAreaColorProfile getColorsProfile() {
		boolean isDarkTheme = UiUtils.isDarkTheme(Objects.requireNonNull(defaultColors.getColor(CodeAreaBasicColors.TEXT_BACKGROUND)));
		Color markAllHighlightColor = isDarkTheme ? Color.decode("#32593D") : Color.decode("#ffc800");
		Color currentMatchColor = defaultColors.getColor(CodeAreaBasicColors.SELECTION_BACKGROUND);
		defaultColors.setColor(CodeAreaMatchColorType.MATCH_BACKGROUND, markAllHighlightColor);
		defaultColors.setColor(CodeAreaMatchColorType.CURRENT_MATCH_BACKGROUND, currentMatchColor);
		return defaultColors;
	}
```
im used replace exits search colors, BasicCodeAreaColorsProfile dont have it